dimager/views.py

```diff
@@ -1,3 +1,109 @@
 from django.shortcuts import render
+from django.contrib.auth.decorators import login_required
+from dimager.models import ImagerProfile
+from imager_images.models import Album, Photo, User
+from django.views.generic import CreateView, UpdateView, DeleteView
 
-# Create your views here.
+
+@login_required()
+def user_profile(request, id):
+    user = User.objects.get(id=id)
+    try:
+        albums = Album.objects.filter(user=user).all()
+    except:
+        albums = None
+    profile = ImagerProfile.objects.get(user=user)
+    num_photos = Photo.objects.filter(user=user).count()
+    return render(request, 'profile.html', {
+        'albums': albums,
+        'user': user,
+        'profile': profile,
+        'num_photo': num_photos
+    })
+
+
+@login_required()
+def user_profile(request):
+    user = request.user
+    try:
+        albums = Album.objects.filter(user=user).all()
+    except:
+        albums = None
+    profile = ImagerProfile.objects.get(user=user)
+    num_photos = Photo.objects.filter(user=user).count()
+    return render(request, 'profile.html', {
+        'albums': albums,
+        'user': user,
+        'profile': profile,
+        'num_photo': num_photos
+    })
+
+
+@login_required()
+def library(request):
+    user = request.user
+    try:
+        albums = Album.objects.filter(user=user).all()
+    except:
+        albums = None
+    try:
+        photos = Photo.objects.filter(user=user).all()
+    except:
+        photos = None
+    return render(request, 'library.html', {
+        'user': user,
+        'albums': albums,
+        'photos': photos
+    })
+
+
+@login_required()
+def stream(request):
+    user = request.user
+    try:
+        albums = Album.objects.filter(user=user).order_by('-date_created')
+    except:
+        albums = None
+    try:
+        followed_albums = Album.objects.filter(user=user.
+                                               profile.is_following()).order_by('-date_created')
+    except:
+        followed_albums = None
```

Comment:
You should be able to come up with a single query that returns you a single list that contains **both** your own albums **and** those owned by the people you are following.  It would be a good place to use `Q` objects.